tripy/tripy/frontend/trace/ops/where.py

```diff
@@ -53,6 +53,10 @@ def infer_shape_output_idxs(self, inputs):
                 ]
             )
 
+    def infer_len(self):
+        # broadcast to the largest of the input lengths
+        return [max(map(lambda inp: op_utils.get_trace_shape(inp)[0], self.inputs))]
```

Comment:
@slyubomirsky looks like we are calling `get_trace_shape` in almost all `infer_len` implementations, isn't this very expensive? (at-least with the current implementation of `get_trace_shape` where we are recursing back to inputs in almost every case).

Reply:
It can be expensive but we do need to know the input shapes in these cases. In some cases (e.g., constants), the shape is set directly so `get_trace_shape` would not have to call into the `ShapeContext`. I do not see any way to avoid it besides doing shape inference in Tripy (which we had reasons for avoiding).